pkg/config/validation/validation.go

```diff
@@ -3321,6 +3321,13 @@ var ValidateServiceEntry = registerValidateFunc("ValidateServiceEntry",
 				errs = appendValidation(errs, fmt.Errorf("invalid host %s", hostname))
 			} else {
 				errs = appendValidation(errs, ValidateWildcardDomain(hostname))
+				for _, port := range serviceEntry.Ports {
+					if port.Protocol == string(protocol.HTTPS) {
```

Comment:
only https protocol does not support partial wildcard?

Reply:
correct. That is what gets translated to server_names in Envoy. I do not know of any other such restrictions @hzxuzhonghu do you know?

Reply:
Even if we can I don't think its good to support them. However, we can't break compat if it used to work, so maybe we can just put a warning

Reply:
Sorry. Confused. What is the conclusion? This specific case can be error because it should have never worked. Are you suggesting for other cases we should add warning?

Reply:
I mean prohibitate wildecard hostname is perhaps too strict. 

If a service has related VS with tls set, it is allowed. You can test it

Reply:
Ok. Moved to warning. Should validate and skip such type of services [here](https://github.com/istio/istio/blob/3ac9654ed50c7e4c387c97b3e12060f91bf1801a/pilot/pkg/networking/core/v1alpha3/tls.go#L203) - May be not worth complicating that stuff?